config/config.go

```diff
@@ -14,21 +14,76 @@
 
 package config
 
+import (
+	"errors"
+	"fmt"
+)
+
+var (
+	ErrUsernamePasswordMissing = errors.New("username or password missing")
+)
+
 type Config struct {
 	// Host of the Weaviate instance.
 	Endpoint string `json:"endpoint" validate:"required"`
 
 	// Scheme of the Weaviate instance.
 	Scheme string `json:"scheme" default:"https" validate:"inclusion=http|https"`
 
-	//TODO: support additional auth schemes __sL__
-	// https://github.com/conduitio-labs/conduit-connector-weaviate/issues/4
-
-	// A Weaviate API key
-	APIKey string `json:"apiKey"`
+	Auth Auth `json:"auth"`
 
 	// The class name as defined in the schema.
 	// A record will be saved under this class unless
 	// it has the `weaviate.class` metadata field.
 	Class string `json:"class" validate:"required"`
 }
+
+func (c *Config) Validate() error {
+	return c.Auth.Validate()
+}
+
+type Auth struct {
+	// Mechanism specifies in which way the connector will authenticate to Weaviate.
+	Mechanism string `json:"mechanism" validate:"inclusion=none|apiKey|wcsCreds" default:"none"`
+
+	// A Weaviate API key.
+	APIKey string `json:"apiKey"`
+
+	// Weaviate Cloud Services (WCS) credentials.
+	WCSCredentials WCSCredentials `json:"wcs"`
```

Comment:
last comment I promise 👀
I think it's better for this json value to match the mechanism name, so it won't be confusing for the user.
so either name this `wcsCreds` or rename the mechanism to be `wcs`, wdyt?

Reply:
No problems, all suggestions are welcome!

You're making a good point. I've stayed from calling the mechanism WCS, because it's not really that (it's simply their cloud service). The authentication is made using WCS credentials.

Now, when it comes to this field: `auth.wcs.username` read good and short to me. And it also technically makes sense: you're **auth**enticating with your **WCS username**. 

On the other hand, being consistent is also good. I don't feel strongly this way or the other, I can change to `wcsCreds`.

Reply:
yes! I knew what you were thinking when naming it, and I think I would've had a problem choosing too.
I don't feel strongly about it either, we can keep it as is, cuz it does make sense :D

Reply:
I've changed it in 1b7bfb8 to `wcsCreds` everywhere, so that it's consistent. If there's overwhelming feedback for something, we can easily change it.